pkg/reconciler/managed/reconciler.go

```diff
@@ -445,9 +448,10 @@ type Reconciler struct {
 	client     client.Client
 	newManaged func() resource.Managed
 
-	pollInterval        time.Duration
-	timeout             time.Duration
-	creationGracePeriod time.Duration
+	pollInterval              time.Duration
+	timeout                   time.Duration
+	creationGracePeriod       time.Duration
+	managementPoliciesEnabled bool
```

Comment:
I wonder if we should just pass some `feature.Flags` to the reconciler? Doesn't make much difference now but it would be handy if we ever end up having multiple feature flags that affect it.

Reply:
I have mixed feelings about this. Currently, AFAIK, we have two feature flags affecting the runtime code for this and external secret stores. While it makes sense to pass the feature flag for management policies, external secret store is a bit different since we would need to pass [another parameter](https://github.com/turkenh/upbound-provider-aws/blob/cd9a184a48b63e3bc4f36a35e3c8c08e8a6d829b/internal/controller/ec2/vpc/zz_controller.go#L34) other than the flag itself.

The most significant advantage would be not needing to update all Setup calls when we add a new feature flag, but this does not hold for 1 of the 2 examples we have today.  

[This](https://github.com/turkenh/upbound-provider-aws/blob/cd9a184a48b63e3bc4f36a35e3c8c08e8a6d829b/internal/controller/ec2/vpc/zz_controller.go#L32-L51) is how we are handling feature flags for those two features today, do you still think it makes sense?

Reply:
One other thought here - this will need to default to enable once we go beta (and the feature is still behind a flag, but on by default). I don't think that would impact this decision though (presumably it could just leave this code as is, and change the default flag value in the callers to enabled).